coop_membership/migrations/12.0.1.1.1/post-migrate_partner_bod_gender.py

```diff
@@ -0,0 +1,4 @@
+def migrate(cr, version):
+    if not version:
+        return
+    cr.execute("update res_partner set birthdate_date=birthdate")
```

Comment:
Have you tested the migration script?

I'm concerned that maybe the column is not there anymore when it's ran.
Just asking you to double check because if we mess this up the customer will loose birthday information from lots of members.. so we need to be 100% sure.

Can you confirm?

Reply:
Yes, Working! Attached video in Task.
https://www.druidoo.io/web#id=2327&action=367&model=project.task&view_type=form&menu_id=237
Gender field still remains to change.

Reply:
It's not working :(

```
odoo_1  | 2020-05-20 18:05:04,837 517 INFO migrate odoo.modules.migration: module coop_membership: Running migration [12.0.1.1.1>] post-migrate_partner_bod_gender 
db_1    | 2020-05-20 18:05:04.837 UTC [58] ERROR:  column "birthdate" does not exist at character 39
db_1    | 2020-05-20 18:05:04.837 UTC [58] STATEMENT:  update res_partner set birthdate_date=birthdate
odoo_1  | 2020-05-20 18:05:04,838 517 ERROR migrate odoo.sql_db: bad query: update res_partner set birthdate_date=birthdate
odoo_1  | ERROR: column "birthdate" does not exist
odoo_1  | LINE 1: update res_partner set birthdate_date=birthdate
odoo_1  |                                               ^
odoo_1  |  
odoo_1  | 2020-05-20 18:05:04,838 517 ERROR migrate odoo.modules.loading: Error executing post migration script for module coop_membership
odoo_1  | `-> coop_badge_reader
odoo_1  | `-> coop_memberspace
odoo_1  | `-> coop_point_of_sale
odoo_1  | `-> coop_print_badge
odoo_1  | : column "birthdate" does not exist
odoo_1  | LINE 1: update res_partner set birthdate_date=birthdate
odoo_1  |                                               ^
odoo_1  |  
odoo_1  | 2020-05-20 18:05:04,847 517 WARNING migrate odoo.modules.loading: Transient module states were reset 
odoo_1  | 2020-05-20 18:05:04,849 517 ERROR migrate odoo.modules.registry: Failed to load registry 
odoo_1  | Traceback (most recent call last):
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/registry.py", line 86, in new
odoo_1  |     odoo.modules.load_modules(registry._db, force_demo, status, update_module)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/loading.py", line 480, in load_modules
odoo_1  |     force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/loading.py", line 364, in load_marked_modules
odoo_1  |     upg_registry=upg_registry,
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/loading.py", line 257, in load_module_graph
odoo_1  |     migrations.migrate_module(package, 'post')
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/migration.py", line 199, in migrate_module
odoo_1  |     migrate(self.cr, installed_version)
odoo_1  |   File "/home/odoo/src/repositories/foodcoops/coop_membership/migrations/12.0.1.1.1/post-migrate_partner_bod_gender.py", line 4, in migrate
odoo_1  |     cr.execute("update res_partner set birthdate_date=birthdate")
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/sql_db.py", line 148, in wrapper
odoo_1  |     return f(self, *args, **kwargs)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/sql_db.py", line 225, in execute
odoo_1  |     res = self._obj.execute(query, params)
odoo_1  | psycopg2.ProgrammingError: column "birthdate" does not exist
odoo_1  | LINE 1: update res_partner set birthdate_date=birthdate
odoo_1  |                                               ^
odoo_1  | 
odoo_1  | 2020-05-20 18:05:04,850 517 CRITICAL migrate odoo.service.server: Failed to initialize database `migrate`. 
odoo_1  | Traceback (most recent call last):
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/service/server.py", line 1151, in preload_registries
odoo_1  |     registry = Registry.new(dbname, update_module=update_module)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/registry.py", line 86, in new
odoo_1  |     odoo.modules.load_modules(registry._db, force_demo, status, update_module)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/loading.py", line 480, in load_modules
odoo_1  |     force, status, report, loaded_modules, update_module, models_to_check, upg_registry)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/loading.py", line 364, in load_marked_modules
odoo_1  |     upg_registry=upg_registry,
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/loading.py", line 257, in load_module_graph
odoo_1  |     migrations.migrate_module(package, 'post')
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/modules/migration.py", line 199, in migrate_module
odoo_1  |     migrate(self.cr, installed_version)
odoo_1  |   File "/home/odoo/src/repositories/foodcoops/coop_membership/migrations/12.0.1.1.1/post-migrate_partner_bod_gender.py", line 4, in migrate
odoo_1  |     cr.execute("update res_partner set birthdate_date=birthdate")
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/sql_db.py", line 148, in wrapper
odoo_1  |     return f(self, *args, **kwargs)
odoo_1  |   File "/home/odoo/.local/lib/python3.5/site-packages/odoo/sql_db.py", line 225, in execute
odoo_1  |     res = self._obj.execute(query, params)
odoo_1  | psycopg2.ProgrammingError: column "birthdate" does not exist
odoo_1  | LINE 1: update res_partner set birthdate_date=birthdate
odoo_1  |                                               ^
```

Reply:
Can you please tell me when you are getting an issue? means during the upgrade module?
Because I just check it with the test database, it seems working as well as updating the birthdate field value.
I am not getting any issues.